stag-library/src/test/java/com/vimeo/stag/KnownTypeAdaptersTestKt.kt

```diff
@@ -37,6 +38,11 @@ class KnownTypeAdaptersTestKt {
         KnownTypeAdapters.PrimitiveByteTypeAdapter.read(JsonReader(StringReader("test".quote())), 1)
     }
 
+    @Test
+    fun `PrimitiveByteTypeAdapter is not instantiable`() {
+        testZeroArgumentConstructorFinalClass(KnownTypeAdapters.PrimitiveByteTypeAdapter::class.java)
```

Comment:
You could make `T` `reified` in `testZeroArgumentConstructorFinalClass` so that this can can be written a bit nicer (`testZeroArgumentConstructorFinalClass<KnownTypeAdapters.PrimitiveByteTypeAdapter>()`)

eg, something like:
```kotlin
inline fun <reified T> testZeroArgumentConstructorFinalClass() {
    assertThatThrownBy {
        T::class.java.getDeclaredConstructor().apply {
            isAccessible = true
        }.newInstance()
    }.hasCauseInstanceOf(UnsupportedOperationException::class.java)
}
```